Tips 'n tricks/Swift-Objective C interoperability and best practices.md

```diff
@@ -134,6 +133,83 @@ If you need to use Swift code in `.h` file, you should use something called Forw
 
 ![1](/img/iOS-objc-swift-interoperability.png)
 
+### Swift enums and Objective-C
+
+#### Header Files - `.h`
+
+>IMPORTANT: Swift enums can't be used in Objective-C header files.
+
+But if your project must have Swift enums in header files, there are two possible solutions:
+
+ 1. Write Swift enum in Objective-C and use newly written enum in Swift files. By doing this, you will have codebase that is easy to maintain.
+
+ 2. Have enum written both in Objective-C and Swift.
+
+ Although having duplicated code is inherently a bad thing, and it will be more challenging to maintain the codebase, using this solution your code will become more flexible and here are the perks that come with it:
+   * You will avoid 'dirtying up' your Swift codebase with Objective-C enums. And by doing this,  somewhere in the future when your project won't use Objective-C anymore, you won't have to go through Swift files and refactor them.
+   * If your project is a library which is used in other projects and has support for both Objective-C and Swift, you won't force Swift developers to have support for Objective-C (bridging headers), and their project can remain pure Swift.
```

Comment:
```suggestion
   * If you want to share your code with others, you won't force Swift developers to have support for Objective-C (bridging headers), and their project can remain pure Swift.
```

Every dependency manager will handle this automatically (i.e. you can use lib written in Obj-C in Swift project). But if you want to inject your ObjC file(s) into the Swift project, then you'll need to add bridging header.